### PR TITLE
docker: upgrade rust version to date `2016-09-29`

### PR DESCRIPTION
**Pull request by iamxy**
PTAL @siddontang

### CODE REVIEW

**Comment by siddontang** [comment]
LGTM

File: docker/rust_dockerfile [hunk 1]
```diff
@@ -25,4 +25,4 @@ RUN cd / && \
     cd / && \
     rm -rf /rocksdb-4.12.fb /rocksdb.tar.gz
 
-RUN curl -sSf https://static.rust-lang.org/rustup.sh | sh -s  -- --date=2016-09-05 --disable-sudo -y --channel=nightly
+RUN curl -sSf https://static.rust-lang.org/rustup.sh | sh -s  -- --date=2016-09-29 --disable-sudo -y --channel=nightly
```
Comment (BusyJay) [inline docker/rust_dockerfile]: Why not upgrade to `2016-10-04`?

Comment (iamxy) [inline docker/rust_dockerfile]: I've upgraded rust to `2016-09-26` in jenkins-CI's image. This image just keeps pace with the version in CI

**Comment by iamxy** [comment]
@BusyJay , I've upgraded rust to `2016-10-04`, both in CI image and this Dockerfile.
PTAL

**Comment by BusyJay** [comment]
LGTM

**Comment by iamxy** [comment]
upgraded rust ver to `2016-10-06`

**Pull request merged**